generator/internal/discovery_type_vertex.cc

```diff
@@ -13,12 +13,28 @@
 // limitations under the License.
 
 #include "generator/internal/discovery_type_vertex.h"
+#include "generator/internal/codegen_utils.h"
 #include "google/cloud/internal/absl_str_join_quiet.h"
+#include "google/cloud/internal/make_status.h"
 #include "absl/strings/str_format.h"
+#include "absl/types/optional.h"
 
 namespace google {
 namespace cloud {
 namespace generator_internal {
+namespace {
+
+absl::optional<std::string> CheckForScalarType(nlohmann::json const& j) {
+  std::string type;
+  if (j.contains("type")) type = j["type"];
```

Comment:
Nit: You could use `value()` again, to avoid the double lookup, and (I believe) make it a little clearer.
```
  std::string type = j.value("type", "");
```